### ISSUE TITLE
[Task] Consider using `GITHUB_TOKEN` in submodule workflow

### ISSUE THREAD

**Issue by ebaron** [cryostatio/cryostat-web#1108]
If we can use `GITHUB_TOKEN` with write access to update Cryostat's submodule we should do that instead of using a Personal Access Token. This would also be useful for the similar workflow in the Helm Chart.

**Comment by ebaron** [comment]
This may not be possible since in this case we're writing to a different repo within the org, but it's worth exploring to see if it's doable.

**Comment by tthvo** [comment]
I guess a solution can be:

- Add a workflow on target repository that will automatically merge when PR is labeled with `auto-merge` (maybe also need automatic approvals).
- Open the PR (instead of direct push) to update submodule with label `auto-merge`.

**Comment by mwangggg** [comment]
I don't think this is possible right now without a `PAT` due to the limitations of the `GITHUB_TOKEN` permissions. Since we want to keep it automated, I'll just leave it as is :)
